### PR TITLE
VEN-1206, VEN-1239 | Create and send berth switch offers

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Offer refactoring
  * Renamed `offer` components to `berthOffer`,
  * Refactored `berthOffer` components, extracted `BerthOfferTable`
  * Moved `switchPlace` to `berthOffer`
  * Split `BerthOfferWithoutApplication` from `BerthOffer`
* Create berth switch offers
  * Note: The application status does not change correctly yet
* Added a modal for selecting which lease to replace
* Disable creating multiple berth switch choices (`berthSwitchOffered`)
* Send berth switch offers
  * Note: The mutation is called, but nothing happens

### Closes :no_good_woman:

* [VEN-1206](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1206): FE - Send a berth switch offer
* [VEN-1239](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1239): Add dialog for exhchange application handling when system can't match pier/berth data

### Automated tests :gear:️

* TODO?

### Manual testing :construction_worker_man:

* TODO

## Screenshots :camera_flash:
image
image

### CODE REVIEW

File: src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    { prop: wasteCollection, key: 'wasteCollection', icon: IconTrash },
+    { prop: electricity, key: 'electricity', icon: IconPlug },
+    { prop: lighting, key: 'lighting', icon: IconStreetLight },
+    { prop: gate, key: 'gate', icon: IconFence },
+    { prop: water, key: 'water', icon: IconWaterTap },
+  ];
+
+  return (
+    <>
+      <Card className={classNames(className)}>
```
Comment (Nizar-Rahme) [inline src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx:120]:
```suggestion
      <Card className={className}>
```

File: src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        });
+      }
+    });
+  };
+
+  const isNotNull = (property: boolean | null): property is boolean => property !== null;
+  const mapPlaceProperties = (placeProperties: PlaceProperty[]) =>
+    placeProperties.map(
+      ({ prop, key, icon }) =>
+        isNotNull(prop) && (
```
Comment (Nizar-Rahme) [inline src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx:100]:
```suggestion
       prop ?? (
```

File: src/features/berthOffer/BerthSwitchOfferContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  const handleClickSelect = (berth: BerthData) => {
+    setSelectedBerth(berth);
+    createBerthSwitchOffer(berth)
+      .then((result) => {
+        if (result.errors) return;
+        history.goBack();
+        showBerthSuccessToast(berth, t);
+      })
+      .catch(() => {
+        setShowSelectLease(true);
```
Comment (Nizar-Rahme) [inline src/features/berthOffer/BerthSwitchOfferContainer.tsx:94]: Why do we want to set this to `true` when we get an error?

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
I left a couple of nitpicks. Otherwise, fantastic job!

File: src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx (hunk 2 above)
Comment (TatuArvela) [reply src/features/applicationView/berthSwitchOfferCard/BerthSwitchOfferCard.tsx:100]: The logic is backwards, but thanks for the suggestion. Going with this:
```
        prop !== null && (
```

File: src/features/berthOffer/BerthSwitchOfferContainer.tsx (hunk 3 above)
Comment (TatuArvela) [reply src/features/berthOffer/BerthSwitchOfferContainer.tsx:94]: :derp: Indeed, good catch!

Comment (TatuArvela) [reply src/features/berthOffer/BerthSwitchOfferContainer.tsx:94]: Actually, this is for the case when the backend can't figure out what the existing lease is supposed to be. The admin then selects whichever they think is the correct one

**Review by Nizar-Rahme — APPROVED** [review]

**Pull request merged**